mesonbuild/modules/windows.py

```diff
@@ -67,11 +67,16 @@ def compile_resources(self, state, args, kwargs):
             suffix = 'o'
         if not rescomp.found():
             raise MesonException('Could not find Windows resource compiler %s.' % ' '.join(rescomp.get_command()))
-        res_kwargs = {'output': '@BASENAME@.' + suffix,
-                      'arguments': res_args}
-        res_gen = build.Generator([rescomp], res_kwargs)
-        res_output = res_gen.process_files('Windows resource', args, state)
-        return ModuleReturnValue(res_output, [res_output])
+
+        res_kwargs = {
+            'output': '@BASENAME@.' + suffix,
+            'input': args,
+            'command': [rescomp] + res_args,
+        }
+
+        res_target = build.CustomTarget('Windows resource', state.subdir, state.subproject, res_kwargs)
```

Comment:
have you tried generating two resource files in the same project with this patch applied? CustomTarget names must be unique, so I think you probably need to do something like:
```meson
res_target = build.CustomTarget('Windows Resources: ' + res_kwargs['output'], state.subdir, state.subproject, res_kwargs)
```

Reply:
Good point, I have not. Currently trying to make this work. It's more involved than simply appending res_kwargs['output'] to it, though, since that is just a template. I'm not sure how to properly transform it into the real output path there yet.